wp-content/civi-extensions/civirazorpay/api/v3/Civirazorpay/CivicrmRazorpayFetchSettlementDateCron.php

```
@@ -0,0 +1,120 @@
+    ]);
+    
+    \Civi::log()->debug("Payment processor retrieved: ID {$paymentProcessor['id']}");
+    $razorpayProcessor = new CRM_Core_Civirazorpay_Payment_Razorpay('live', $paymentProcessor);
+    $api = $razorpayProcessor->initializeApi();
+    \Civi::log()->debug("Razorpay API initialized successfully");
+
+    // Handle the input date
+    $targetDate = !empty($params['date']) ? $params['date'] : date('Y-m-d', strtotime('-1 day'));
+    \Civi::log()->debug("Raw input date: {$params['date']}, normalized target date: {$targetDate}");
+    
+    $dateTime = DateTime::createFromFormat('Y-m-d', $targetDate);
```

PHPMD (2.15.0)

Avoid using static access to class 'DateTime' in method 'civicrm_api3_civirazorpay_civicrmrazorpayfetchsettlementdatecron'. (Clean Code Rules)

(StaticAccess)